syscalls_darwin.go

```diff
@@ -75,6 +77,19 @@ func openDevSystem(config Config) (ifce *Interface, err error) {
 	if config.DeviceType != TUN {
 		return nil, errors.New("only tun is implemented for SystemDriver, use TunTapOSXDriver for tap")
 	}
+
+	ifIndex := -1
+	if config.Name != "" {
+		const utunPrefix = "utun"
+		if !strings.HasPrefix(config.Name, utunPrefix) {
+			return nil, fmt.Errorf("Interface name must be utun[0-9]+")
+		}
+		ifIndex, err := strconv.Atoi(config.Name[len(utunPrefix):])
```

Comment:
Just tested it, and this actually needs to be 

```suggestion
		ifIndex, err = strconv.Atoi(config.Name[len(utunPrefix):])
```

Otherwise the `ifIndex` gets shadowed and after this block it remains `-1`.